Add: added kerberos related connection fields(principal, keytab)

seyoon-lim opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

softyoungha commented on airflow/providers/apache/spark/hooks/spark_submit.py:
```diff
@@ -268,6 +291,12 @@ def _resolve_connection(self) -> dict[str, Any]:
                 )
             conn_data["spark_binary"] = self.spark_binary
             conn_data["namespace"] = extra.get("namespace")
+            conn_data["principal"] = self._principal if self._principal else extra.get("principal")
```
```suggestion
            conn_data["principal"] = self._principal or extra.get("principal")
```

softyoungha commented on airflow/providers/apache/spark/hooks/spark_submit.py:
```diff
@@ -268,6 +291,12 @@ def _resolve_connection(self) -> dict[str, Any]:
             conn_data["spark_binary"] = self.spark_binary
             conn_data["namespace"] = extra.get("namespace")
+            conn_data["principal"] = self._principal if self._principal else extra.get("principal")
+            base64_keytab = extra.get("keytab")
+            if self._keytab is not None:
+                conn_data["keytab"] = self._keytab
+            elif base64_keytab is not None:
+                conn_data["keytab"] = self._get_keytab_from_base64(base64_keytab)
```
```suggestion
            conn_data["keytab"] = self._keytab
            base64_keytab = extra.get("keytab")
            if base64_keytab is not None:
                conn_data["keytab"] = self._get_keytab_from_base64(base64_keytab)
```

softyoungha reviewed (approved):
👍

softyoungha commented on airflow/providers/apache/spark/hooks/spark_submit.py:
```diff
@@ -281,10 +310,57 @@ def _resolve_connection(self) -> dict[str, Any]:
+            self.log.error("Failed to decode base64 keytab: %s", err)
+            raise AirflowException("Failed to decode base64 keytab") from err
+
+        # validate exists keytab file
+        if keytab_path.exists():
+            with open(keytab_path, "rb") as f:
+                existing_keytab = f.read()
+            if existing_keytab == keytab:
+                self.log.info("Keytab file already exists and is the same as the provided keytab")
+                return str(keytab_path)
+
+        # write new keytab file
+        try:
+            with open(staging_path, "wb") as f:
+                self.log.info("Saving keytab to %s", staging_path)
+                f.write(keytab)
+
+            self.log.info("Moving keytab from %s to %s", staging_path, keytab_path)
+            shutil.move(str(staging_path), str(keytab_path))
+            return str(keytab_path)
```
요거 요런 방법도 있네여

![image](https://github.com/user-attachments/assets/93175aab-983d-4fba-bb42-a7e3e4e0c741)

seyoon-lim replied:
@softyoungha 
그 방법도 고민했는데, 결국 쓰는 도중에 파일을 read하면 문제가 있다고 생각해서

random한 이름으로 staging파일 쓰고, atomic한 mv로 처리하는 식으로 구현했어용
(요 방법이 고민했을때는 제일 안전하다고 판단했슴당 ㅎㅎ)

raphaelauv committed:
remove soft_fail part2 (#41727)

Co-authored-by: raphaelauv <raphaelauv@users.noreply.github.com>

ephraimbuddy committed:
Separate FAB migration from Core Airflow migration (#41437)

* Separate FAB migration from Core Airflow migration

This PR separates FAB migration from Airflow Core migration and provides
a way for apps to integrate into Airflow and run their migrations.

* minor fixes and use pre-commit on fab migrations

* support external db managers to drop tables when airflow db reset is run

* Add tests

* fix autogenerate and update function signatures

* Rename alembic directory to migrations

* add more tests

* Fix import

* Fix static check

* Fix tests

* fixup! Fix tests

* fixup! fixup! Fix tests

* Mark extra operator links test as non-db test

* update erd diagram

* import airflow settings at point of use

* skip import error for alembic env.py

* Remove type on typing

* Move code around

* fixup! Move code around

* Move import

* fixup! Move import

* Don't disable existing loggers

* Use abstract base and update tests

* Revert compat change

* Fix backcompat for fab provider

* Build fab provider for www test instead of installing from pypi

* fixup! Fix backcompat for fab provider

* Mark downgradedb method as abstract

* Ensure supports_table_dropping works and improve tests

* Skip tag check when preparing fab provider in CI

raphaelauv committed:
revert removing soft_fail from TimeDeltaSensorAsync (#41741)

Co-authored-by: raphaelauv <raphaelauv@users.noreply.github.com>

Lee-W committed:
docs(deferring): fix wrong example and remove unnecessay example (#41691)

uranusjr committed:
Unify DAG schedule args and change default to None (#41453)

vatsrahul1001 committed:
Update Databricks workflow example DAG (#41700)

* update databricks workflow example dag

* Format commentted code

* Update tests/system/providers/databricks/example_databricks_workflow.py

* Rearrange comment block

* removing comment

---------

Co-authored-by: Pankaj Koti <pankajkoti699@gmail.com>

dependabot[bot] committed:
Bump micromatch from 4.0.5 to 4.0.8 in /airflow/www (#41726)

Bumps [micromatch](https://github.com/micromatch/micromatch) from 4.0.5 to 4.0.8.
- [Release notes](https://github.com/micromatch/micromatch/releases)
- [Changelog](https://github.com/micromatch/micromatch/blob/4.0.8/CHANGELOG.md)
- [Commits](https://github.com/micromatch/micromatch/compare/4.0.5...4.0.8)

---
updated-dependencies:
- dependency-name: micromatch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

jscheffl committed:
Remove Airflow 2.7 support from Kubernetes Provider (#41746)

dirrao committed:
Provider fab auth manager deprecated methods removed (#41720)

pierrejeambrun committed:
remove licence from generated ts types interface (#41751)

collinmcnulty committed:
Don't Fail LocalTaskJob on heartbeat (#41704)

* Never fail an ltj over a heartbeat

* Log a warning on failed heartbeat

* Avoid using f-string in log

* Remove unnecessary pass statement

amoghrajesh committed:
Adding rel property to hyperlinks in logs (#41696)

* Adding rel property to hyperlinks in logs

* fixing tests

KrunchMuffin committed:
Update elasticsearch.rst (#41764)

Changed Azure to Elasticsearch

JDarDagran committed:
Update snowflake naming for account names and locators. (#41775)

Signed-off-by: Jakub Dardzinski <kuba0221@gmail.com>

kongdewen committed:
[redis]adding additional values for existing persistence claim (#41619)

* Update values.yaml

* Update redis-statefulset.yaml

* Update values.schema.json

* Update redis-statefulset.yaml

* add testing

* update unit testing

* update

hoalongnatsu committed:
chore(docs): add an example for auth with keycloak (#41687)

* chore(docs): add an example for auth with keycloak

Added a new section in the authentication documentation that provides a code of configuring Airflow to work with Keycloak.

* chore(docs): add an example for auth with keycloak

Fix spelling and styling

* chore(docs): add an example for auth with keycloak

Fix static checks

dirrao committed:
Removed deprecated method referance airflow.www.auth.has_access when min airflow version >= 2.8.0 (#41747)

jscheffl committed:
Remove deprecations in airflow.models.skipmixin (#41780)

* Remove deprecations in airflow.models.skipmixin

* Fix pytests

jscheffl committed:
Remove deprecations in airflow.models.errors (#41779)

* Remove deprecations in airflow.models.errors

* Fix static code checks in compat

uranusjr committed:
Fix some non-critical pre-commit issues (#41749)

The '.venv' name (common for in-project virtual environments) is not
included in rat-excludes, causing a lint failure since a license is not
found in it. Since 'venv' (no leading dot) is excluded, let's also add
'.venv' to the file.

The 'downgradedb' method of BaseDBManager is declared abstract, but not
implemented in FABDBManager. This causes Mypy to error when FABDBManager
is instantiated. We don't always need the implement the method, so let's
remove the abstractmethod decorator.

Finally (and least critically), when you run pre-commit during a merge,
a hook checks if you're committing unresolved conflicts. The check is
not particularly smart, however, and can mis-identify header underlines
in restructuredText as false positives. This is easy to fix; just add
one more character to the underline. It's fine as long as the underline
is not EXACTLY SEVEN characters.

jscheffl committed:
Remove deprecations in airflow.models.param (#41776)

* Remove deprecations in airflow.models.param

* Fix pytests

geraj1010 committed:
Add logging device and logging device options to DockerSwarmOperator (#41416)

jscheffl committed:
Remove deprecations in Connection for Airflow 3 (#41762)

* Remove deprecations in Connection for Airflow 3

* Fix pytests

jscheffl committed:
Remove deprecations in BaseOperator for Airflow 3 (#41761)

* Remove deprecations in BaseOperator for Airflow 3

* Fix pytests

jscheffl committed:
Remove a set of deprecations in airflow.www module (#41758)

jscheffl committed:
Remove one leftover deprecated import from FAB provider test (#41757)

jscheffl committed:
Removed deprecated TaskStateTrigger from airflow.triggers.external_task module (#41737)

jscheffl committed:
Remove deprecated parameters from airflow (core) Operators (#41736)

jscheffl committed:
Remove deprecated airflow.kubernetes package (#41735)

jscheffl committed:
Remove deprecation warning for cgitb in Plugins Manager (#41732)

dirrao committed:
backfill job command cli deperecated options removed (#41739)

* backfill job command cli deperecated options removed

* news fragment added

dirrao committed:
Module airflow.hooks.dbapi removed (#41748)

* Deprecated module airflow.hooks.dbapi removed

* news fragment added

jscheffl committed:
Remove deprecations in airflow.models.dagrun (#41778)

* Remove deprecations in airflow.models.dagrun

* Update newsfragments/41778.significant.rst

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

---------

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

jscheffl committed:
Remove deprecated get_connections() function in BaseHook (#41733)

* Remove deprecated get_connections() function in BaseHook

* Adding newsfragment after PR open

dirrao committed:
airflow.models.taskinstance deprecations removed (#41784)

potiuk committed:
Fix mocking in cncf.kubernetes tests after get_connections removal (#41805)

The #41733 removed deprecated get_connections but it was still
used in cncf.kubernetes tests.

Lee-W committed:
test(providers/yandex): fix test_yandex_lockbox_secret_backend_get_connection_from_json by removing non-json extra (#41815)

eladkal committed:
Add examples for discussion/vote threads for new providers (#41817)

utkarsharma2 committed:
Fix: DAGs are not marked as stale if the dags folder change (#41433)

* Fix: DAGs are not marked as stale if the AIRFLOW__CORE__DAGS_FOLDER changes

* Update airflow/dag_processing/manager.py

* Add testcase

* Add code comment

* Update code comment

* Update the logic for checking the current dag_directory

* Update testcases

* Remove unwanted code

* Uncomment code

* Add processor_subdir when creating processor_subdir

* Fix test_retry_still_in_executor test

* Remove config from test

* Update airflow/dag_processing/manager.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

* Update if condition for readability

---------

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

eladkal committed:
Update providers metadata 2024-08-28 (#41823)

jscheffl committed:
Move away from deprecated DAG.following_schedule() method (#41773)

* Move away from deprecated DAG.following_schedule() method

* Update mock to changed logic and remove one layer of Pendulum

VladaZakharova committed:
Small fix for datafusion system test (#41795)

tirkarthi committed:
Set end_date and duration for triggers completed with end_from_trigger as True. (#41754)

HassanAlahmed committed:
Add hostAliases support for Triggerer in helm chart (#41725)

gagan-bhullar-tech committed:
logout link in no roles error page fix (#41813)

* logout link in no roles error page fix

* review comments

marcosmarxm committed:
Update provider to use Airbyte API Python SDK (#41122)

* rename fields login=>client-id and password=>client-secret

* replace HttpHook and use airbyte-api sdk

* fix unit tests for op, hook, sensor and triggers

* run precommit and fix files

* fix breeze unit test selective checks

* remove skipPolicy

* remove skipPolicy

* make get_job_status test async

* add async pytest

* remove async function

* make airbyte_api dependency >= 0.51.0 latest version

* regenerate provider_dependencies.json

* remove http provider dependency and update changelog

* refactor execute op function

* Update airflow/providers/airbyte/CHANGELOG.rst

* make to work with OSS and any generic FQDN addres

* regenerate provider_dependencies.json

* fix tests

* regenerate provider_dependencies.json

* solve conflict changelog

* solve conflict dependencies

* fix conflicts and run-precommit

* rename function get_conn

* readd line in changelog

* add pytest db_test

* add db_Test to class level

* add pytest to op test class

* fix sensor conflict

---------

Co-authored-by: Elad Kalif <45845474+eladkal@users.noreply.github.com>

darth-drew committed:
Consolidated fix for volumeClaimTemplates (apiVersion and PVC) (#41771)

jscheffl committed:
Remove deprecations in airflow.models.dag (#41774)

jscheffl committed:
Remove deprecations in airflow.models.taskreschedule (#41808)

* Remove deprecations in airflow.models.taskreschedule

* Review feedback

MaksYermak committed:
Make code style fix in Dataform system test (#41750)

uranusjr committed:
Remove old pre-migration data integrity checks (#41850)

utkarsharma2 committed:
Handle Example dags case when checking for missing files (#41856)

Earlier PR create to address the issue was not handling the case for the Example Dags, due to which the example dags were marked as stale since they are not present in the dag_directory. This PR handles that scenarios and update the testcase accordingly.

related: #41432

MaksYermak committed:
Unpin google-cloud-bigquery package version for Google provider (#41839)

pierrejeambrun committed:
Pydantic v2 and FastAPI as core dependency (#41857)

* update requirements

* Pydantic dependancy is not optional anymore

* add newsfragment

* fix ci

* Fix remnants of deleted tests

eladkal committed:
Remove tabular provider from source code (#41826)

dependabot[bot] committed:
Bump webpack from 5.76.0 to 5.94.0 in /airflow/www (#41864)

Bumps [webpack](https://github.com/webpack/webpack) from 5.76.0 to 5.94.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.76.0...v5.94.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pierrejeambrun committed:
Setup ui rest api (#41798)

* init code and basic view

* add more modularization

* factorize dag bag instantiation code

moiseenkov committed:
Refactor test_cli_test_with_deferrable_operator to use caplog (#41859)

GlenboLake committed:
Fix treatment of "#" in S3Hook.parse_s3_url() (#41796)

The current implementation of parse_s3_url will truncate a key if it contains
an octothorpe character. By passing the allow_fragments=False argument to
urlsplit, keys will be correctly parsed.

swythan committed:
Fix typo in trace attribute (#41837)

jabbera committed:
Support the unpack operator in signature (#41316)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: jabbera <github@bary.io>
Co-authored-by: jabbera <michael.barry@gmo.com>

gaurav7261 committed:
[FEAT] databricks repair run with reason match and appropriate new settings (#41412)

pankajkoti committed:
Use Databricks conn_id from env for DatabricksTaskOperator in system test (#41861)

amoghrajesh committed:
Fixing version bump issue when all changes are misc (#41875)

amoghrajesh committed:
Adding tojson filter to example_inlet_event_extra example dag (#41873)

MaksYermak committed:
Update min version of google-cloud-bigquery package (#41882)

howardyoo committed:
Add trace tags to task instances and also support OTEL_RESOURCE_ATTRIBUTES for spans (#41841)

CYarros10 committed:
Add Supervised Fine Tuning Train Operator, Hook, Tests, Docs (#41807)

* add supervised_fine_tuning

* build fix

* build,test fix

* unit test build fix

* xcom fix

* refactor supervised tuning into generative_model module, PR feedback, tests

* minor system test fix

* update provider.yaml

* doc fix

* Update Vertex AI Documentation

utkarsharma2 committed:
Skip test_scan_stale_dags_when_dag_folder_change in DB isolation mode (#41893)

Since the similar test(test_scan_stale_dags_standalone_mode) are skipped in DB isolation mode

bbovenzi committed:
Initialize new airflow/ui directory (#41846)

* Initialize new airflow/ui directory

configure flask blueprint and render real dags list

Add banner to switch back and forth between UIs

Add readme

basic test setup, chakra not working

Move Flask blueprint, build navbar

Switch to react with working tests

Basic hot refresh

add prettier

do not render the new UI banner on loggin page

add pre-commits to ui

add CI integration

update doc accordingly

redirect to login page on 401+403 errors

update lint command

update lint, lint:fix command

Add codegen to pre-commit

Add ui to breeze

Update build commands

* Fix dev favicon and table strip css bleedover

* breeze and CI fixes

* Fix github actions

* Try fixing selective checks and github actions again

* set pnpm cache-dependency-path

* Fix breeze static checks

* fix ci

* Update CONTRIBUTING.md

---------

Co-authored-by: pierrejeambrun <pierrejbrun@gmail.com>

howardyoo committed:
Adding DAG import error stack trace as span event in OTEL traces for Airflow (#41865)

* enhancement for issue #41863

* fixed minor typo

* Update airflow/dag_processing/manager.py

Thanks for the snippet! Will commit this suggestion.

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

* Update manager.py

reformatting

---------

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

jscheffl committed:
Make Scarf usage reporting in major+minor versions and counters in buckets (#41898)

potiuk committed:
Add generating recommended actions for projects and dependency depth (#41920)

The SBON export dependencies command now supports generating
recommended actions and dependency depth information. recommended
actions and dependency depth information.

dirrao committed:
Removed deprecated method requires_access from module airflow.api_connexion.security (#41910)

* Removed deprecated method requires_access from module airflow.api_connexion.security

* news fragment added

Owen-CH-Leung committed:
Fix ElasticSearch SQLClient deprecation warning (#41871)

* Remove importing SQLClient, and use sql interface directly from elasticsearch client

* Fix failing CI test

* Remove checking pytest in modules, and patch ElasticSearch directly instead of SQLClient

moiseenkov committed:
Enforce deprecation message format with EOL for google provider package (#41637)

* Enforce deprecation message format with EOL for google provider package

* A custom @deprecated decorator for Google prvider package.

lukas-mi committed:
update pattern for dataflow job id extraction (#41794)

dirrao committed:
Removed deprecation warning catching in test_db_cleanup (#41909)

Lee-W committed:
ci: improve check_deferrable_default script to cover positional variables (#41924)

potiuk committed:
Upgrade UV to 0.4.1 (#41925)

dstandish committed:
Add backcompat check for executors that don't inherit BaseExecutor (#41906)

HuanjieGuo committed:
[Task] add UT for rotate log setting in worker task (#41923)

dirrao committed:
airflow.models.xcom deprecations removed (#41803)

gopidesupavan committed:
Fix poll_interval in GKEJobTrigger (#41712)

CYarros10 committed:
Add CountTokensOperator for Google Generative AI CountTokensAPI (#41908)

* Add CountTokensOperator for Google Generative AI CountTokensAPI

* Update system test DAG with correct arguments

potiuk committed:
Protect against None components of universal pathlib xcom backend (#41921)

fixes: #41723

seyoon-lim committed:
Merge branch 'main' into feature/add-krb-field-on-spark-conn

seyoon-lim committed:
added additional comment on create_keytab parts

potiuk committed:
Lower down universal-pathlib minimum to 0.2.2 (#41939)

Beacause we pinned universal-pathlib to 0.2.2 in 2.10.0 adding >=
0.2.3 conflicts with currently released PyPI packages. We shoudl
lower it down in order to allow PyPI constraints to not be
conflicting

seyoon-lim committed:
Merge branch 'main' into feature/add-krb-field-on-spark-conn

pierrejeambrun committed:
add airflow ui-api command (#41896)

add port binding

Add tests

Add ui-api to airflow standalone

Fix bad descriptor error on shutting down the worker

moiseenkov committed:
Remove system test for derepcated Google analytics operators (#41946)

The API is turned off and no longer works

CYarros10 committed:
Add RunEvaluationOperator for Google Vertex AI Rapid Evaluation API (#41940)

mobuchowski committed:
fix: cast list to flattened string in openlineage InfoJsonEncodable (#41786)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

gopidesupavan committed:
docstring update (#41929)

mobuchowski committed:
move to dag_run.logical_date from execution date in OpenLineage provider (#41889)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

kacpermuda committed:
chore: Update Marquez version in openlineage integration (#41950)

Signed-off-by: Kacper Muda <mudakacper@gmail.com>

mobuchowski committed:
feat: notify about potential serialization failures when sending DagRun, don't serialize unnecessary params, guard listener for exceptions (#41690)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

Mayuresh16 committed:
[Fix #41763]: Redundant forward slash in SFTPToGCSOperator when destination_path is not specified or have default value (#41928)

* fix the redundant forward issue if the has

* Add a test for fix

* respect airflow standards linting

* update destination path to destination_path in comment for clarity

---------

Co-authored-by: Mayuresh Kedari <Mayuresh.Kedari@datametica.com>

seyoon-lim committed:
Merge branch 'main' into feature/add-krb-field-on-spark-conn

moiseenkov committed:
Add DataflowStartYamlJobOperator (#41576)

* Add DataflowStartYamlJobOperator

* Refactor hook and operator

---------

Co-authored-by: Eugene Galan <ehalan@google.com>

potiuk committed:
Revert "Provider fab auth manager deprecated methods removed (#41720)" (#41960)

This reverts commit b0391838c142bebdf178ba030c45db16b1f1f33b.

potiuk committed:
Switch to using emulation for image building (#41959)

Until we can use ARM hardware to build images with the new ARC
architecture https://github.com/apache/airflow/issues/41935
we switch image building to emulated builds. They are 10x slower
but it's better than nothing.

kaxil committed:
Add TODOs in providers code for Subdag code removal (#41963)

jedcunningham committed:
Remove `--tree` flag from `airflow tasks list` command (#41964)

This also removes `DAG.tree_view` and `DAG.get_tree_view`. The format of
the output from these can be expensive to generate and extremely
large, depending on the DAG. Users are better off using `airflow dag
show` instead.

jedcunningham committed:
Add warning that listeners can be dangerous (#41966)

They are necessarily dangerous, but simply due to the way they are
integrated into Airflow, their side effects can be pretty impactful.

topherinternational committed:
Fix `--ignore`-related clobbering when breeze tests providers (#41967)

seyoon-lim committed:
added base64 decode exception test case

pierrejeambrun committed:
Add views tests (#41957)

potiuk committed:
Move projects metadata for SBOM export to Google Spreadsheet (#41930)

Move the "known" project's metadata to a metadata spreadsheet where
we will be able to maintain it without having to modify the code in
breeze. This allows to regenerate SBOM export any time with updated
metadata and allows it to be edited and maintained manually.

This also means that the CSV option should be removed (since we are
now rely on Google Spreadsheet anyway). The code has been refactored
to move things around betwen cdxgen/sbom/ne projects_google_spreadsheet
modules.

Added rich progress to track the advancements of tasks.

seyoon-lim committed:
added keytab write & move exception test case

seyoon-lim committed:
Merge branch 'main' into feature/add-krb-field-on-spark-conn

mattogburke committed:
Correct scheduled slots documentation and missing open telemetry span (#41899)

* Correct documentation for pool.scheduled_slots metrics

* Add missing pool.scheduled_slots telemetry span

dabla committed:
Generalize caching of connection in DbApiHook to improve performance (#40751)

MaksYermak committed:
Add ENV_ID for job_name in Cloud Run system tests (#41788)

molcay committed:
Add triggered_by field to DAG Run model to distinguish the source of a trigger  (#39165)

* Add triggered_by field to DAG Run model to distinguish the source of a trigger

* Fix conflict
Fix import statement for listener test
Fix missing parameter for triggered_by for new test case

---------

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

uranusjr committed:
Drop execution_date unique constraint on DagRun (#41818)

vincbeck committed:
Move `register_views` to auth manager interface (#41777)

bbovenzi committed:
Update UI contributing docs (#41903)

* Update node_env_setup docs

* fix extra blank lines

* Address PR feedback

* Remove redundancy add a note about vite dev server

gopidesupavan committed:
validate aws service exceptions in waiters (#41941)

potiuk committed:
Fix issue generation failing after deleting a PR (#42001)

Seems that #41416 PR has been somehow (?) deleted and when we try
to generate issue links for it, the issue generation fails (and it
is not an issue as well). This change should workaround that by
skipping the PR.

tirkarthi committed:
Implement pagination for the new home page. (#41905)

* Implement pagination for the home page.

* Add page number buttons.

* Change default number of dags to display to 50.

* Fix imports.

* Fix PR comments.

utkarsharma2 committed:
Handle relative paths when sanitizing URLs (#41995)

* Handle relative paths when sanitizing URLs

In the initial PR(https://github.com/apache/airflow/pull/41665) we didn't handle the relative path in URL which led to issue(https://github.com/apache/airflow/issues/41977). This PR aims at handling the relative path case when sanitizing URLs

* Add PR suggestions

* Update code comment

potiuk committed:
Automate exclusion of providers in lowest-dependency tests (#41991)

The "lowest" dependency tests had "hard-coded" exclusion for
those providers that were excluded in specific python versions.

This change automates it:

* selective checks produce dictionary python -> excluded versions
* the dictionary (as string) is passed to unit tests in actions
* the dictionary (as string) is passed to breeze container
  as EXCLUDED_PROVIDERS variable
* "is_provider_excluded.py" checks if the provider is excluded
  and exits with success if so

Related: #41555

bbovenzi committed:
Add codeowners for new UI (#42010)

harryshi10 committed:
bugfix/livy-set-base-url (#41658)

* base_url by default is empty string, should call  when it takes the default value

* check base_url with if not

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

* test_post_batch_calls_get_conn_if_no_base_url

---------

Co-authored-by: Hua Shi <hua.shi@sensortower.com>
Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

potiuk committed:
Exclude google-cloud-spanner 3.49.0 (#42011)

Google Cloud Spanner library released today has a missing
grpc_intercept module.

See: https://github.com/googleapis/python-spanner/issues/1193

perry2of5 committed:
Add callback to process Azure Service Bus message contents (#41601)

jscheffl committed:
Improve docs on markdown docs in DAGs (#42012)

ephraimbuddy committed:
Add back migrations from 2.7.0 (#41994)

* Add back migrations from 2.7.0

Previously, we removed all these migrations. However, there are concerns that
we should make it possible for people to upgrade from 2.7.0 to 3.0.0. This PR
brings back such possibility

* fixup! Add back migrations from 2.7.0

* fixup! fixup! Add back migrations from 2.7.0

* update erd diagram

* fix static checks

potiuk committed:
Mark test_listener_logs_failed_serialization as flaky (#42022)

Making it stable is tracked in #42020

dabla committed:
Allow custom api versions in MSGraphAsyncOperator (#41331)

potiuk committed:
Refresh images manually can now be done per-platform (#42000)

* Refresh images manually can now be done per-platform

The process of refreshing images manually used to refresh separate
images "per platform" - but assumed we refresh both platforms.

This PR allows to refresh the multi-platform images separately per
ech platform (since currently we only refresh AMD in canary builds,
we can refresh images manually for ARM onlyi using MacBooks)

Co-authored-by: Shahar Epstein <60007259+shahar1@users.noreply.github.com>

marbrex committed:
Update dags.rst (#41999)

Change "Setup and teardown" section's level in the document

dondaum committed:
feat: log client db messages for provider postgres (#40171)

topherinternational committed:
fix mock assertion in Livy test (#42027)

topherinternational committed:
Fix mock assertion in Livy test, for real this time (#42030)

bbovenzi committed:
Refactor new UI Nav (#42009)

* Refactor new UI Nav

* Fix dag icon props

gopidesupavan committed:
Use base aws classes in AWS Glue DataBrew Operators/Triggers (#41848)

bbovenzi committed:
Fix dev_mode check in init_react_ui (#42036)

mobuchowski committed:
test: make flaky OL test wait for future callback done (#42028)

Signed-off-by: Maciej Obuchowski <obuchowski.maciej@gmail.com>

Lee-W committed:
Rewrite how DAG to dataset / dataset alias are stored (#41987)

AetherUnbound committed:
Add note about dataclasses and attrs to XComs page (#42045)

* Add note about dataclasses and attrs to XComs page

The [XCom page](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/xcoms.html) doesn't make explicit reference to `dataclass` and `attrs.define`, although the TaskFlow page does. This adds a note for those looking at the possible limitations of XCom types with a reference to where XCom type constraints are also discussed.

uranusjr committed:
Try to make dataset objects totally unhashable (#42054)

topherinternational committed:
Switch cloudant provider from cloudant library to ibmcloudant library (#41555)

potiuk committed:
Revert "Try to make dataset objects totally unhashable (#42054)" (#42065)

This reverts commit 1c4a00bb4e15de083aea0c2a0ffe14ea97955c70.

Lee-W committed:
ci: auto fix default_deferrable value with LibCST (#41984)

ferruzzi committed:
Fix `GlueDataBrewStartJobOperator` template fields (#42073)

dirrao committed:
airflow 3 deprecations removed from trigger_dagrun operator and serialized_objects (#42062)

VladaZakharova committed:
Add new type of exception to catch timeout (#42064)

kmdupr33 committed:
Update example_python_operator.py (#42075)

ferruzzi committed:
Using dedent in the Sagemaker system test makes the code a little easier to read. (#42076)

utkarsharma2 committed:
Chart: Default airflow version to 2.10.1 (#42069)

vincbeck committed:
Rename `--extras` to `--airflow-extras` from `breeze prod-image build` command (#42077)

seyoon-lim committed:
Merge branch 'main' into feature/add-krb-field-on-spark-conn

seyoon-lim committed:
removed _get_keytab & _get_principal

fritz-astronomer committed:
Add `__doc__` property to `@task` (#42041)


Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>
Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>

topherinternational committed:
Try to make dataset objects totally unhashable, redux (#42066)



---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

saurabhb-dev committed:
Update security_model.rst to clear unauthenticated endpoints exceptions (#42057)

* Update security_model.rst

Fixing confusion on exception to unauthenticated endpoints.

* Update security_model.rst

Fixed grammar and spell mistakes.

* Update docs/apache-airflow/security/security_model.rst

* Update docs/apache-airflow/security/security_model.rst

---------

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

jscheffl committed:
Change compat tests to use new Airflow 2.10.1 (#42084)

uranusjr committed:
Use functools.update_wrapper on task decorator (#42079)

Lee-W committed:
ci(pre-commit): lower minimum libcst version to 1.1.0 for python 3.8 support (#42083)

bugraoz93 committed:
Retry Test Airflow Release Step Multiple Times  (#42043)

pierrejeambrun committed:
Add openapi generation hook (#42098)

potiuk committed:
Limit universal-pathlib below 0.2.4 as it breaks our integration (#42090)

Universal Pathlib 0.2.4 adds extra validation for Paths and our
integration with local file paths Does not work with it. The
0.2.5 version fixes it.

Tracked in https://github.com/fsspec/universal_pathlib/issues/276

HaKkaz committed:
add missing unit tests for mongo providers (#41986)


---------

Co-authored-by: anthony0522 <anthony@sudoresearch.com>

molcay committed:
Fix the system test: dataflow_native_java (#42099)

jx2lee committed:
Add parent_model param in `UploadModelOperator` (#42091)

andres-lowrie committed:
Fix typo in plugins.rst (#42106)

I'm not sure if my change is correct but I'm assuming the word `until` was missing in the sentence.

Otherwise that would mean that the plugin changes are never used right?

dirrao committed:
Deprecated metrics configuration removed (#42088)

* Deprecated metrics configuration removed

* news fragment added

dirrao committed:
Deprecated configuration stalled_task_timeout, task_adoption_timeout and worker_pods_pending_timeout removed (#42060)

* Deprecated configuration stalled_task_timeout, task_adoption_timeout and worker_pods_pending_timeout removed

* news fragment added

potiuk committed:
fix indentation for unit test contribution docs (#42110)

The header levels in the unit test contribution docs have been
messed up.

pierrejeambrun committed:
Fix details tab log url detection (#42104)

* Fix details tab log url detection

* Add test

arnaubadia committed:
Add return_immediately as argument to the PubSubPullSensor class (#41842)

Co-authored-by: Arnau Badia Sampera <arnau.badia.sampera@unacast.com>

DjVinnii committed:
Move DAGs table to a reusable DataTable component (#42095)

* Move DAGs table to a reusable DataTable component

Signed-off-by: Vincent Kling <vkling@vinniict.nl>

* Add missing license

Signed-off-by: Vincent Kling <vkling@vinniict.nl>

* Fix typing

Signed-off-by: Vincent Kling <vkling@vinniict.nl>

---------

Signed-off-by: Vincent Kling <vkling@vinniict.nl>

potiuk committed:
Fix failing compatibility test: importing missing saml for old airflows (#42113)

The Compatibility tests for AWS are failing after recent changes
as they attempt to import saml library before skipping the tests
when the import is missing.

gopidesupavan committed:
Aws executor docs update (#42092)

dstandish committed:
Log info not error in schedule when task has failure (#42116)

You have to think about the context where this is emitted.  It is emitted in the scheduler.  It's not a scheduler error.

bbovenzi committed:
Improve DAGs table UI (#42119)

* Rebase and fix filter pagination

* Add pluralize test

bbovenzi committed:
Move Dags list state to url params (#42124)

* Rebase and fix filter pagination

* Add pluralize test

* Move table state to url params

kacpermuda committed:
chore: bump OL provider dependencies versions (#42059)

Signed-off-by: Kacper Muda <mudakacper@gmail.com>

andrii0yerko committed:
fix typo in Tableau Connection (#42131)

Sever -> Server

potiuk committed:
Make SAML a required dependency of Amazon provider (#42137)

Amazon provider with auth manager requires SAML onelogin import and
it starts to be more and more problematic to skip the related tests
for compatibility.

It seems appropriate to move saml to be a required dependency of
Amazon provider in this case.

Since saml is only used by Amazon provider, we can also safely
remove optional extra for it.

pierrejeambrun committed:
Restructure FastAPI API (#42128)

* Restructure FastAPI API

* Fix CI

mf-hyunsoo-kang committed:
Fix simple typo in the documentation. (#42058)

sc-anssi committed:
Support multiline input for Params of type string in trigger UI form (#40414)

* Add multiline input (textarea) support for Params of type string in trigger UI form

* Use the 'format' attribute in the Param for rendering a multiline text area in the trigger UI form

* Update example DAG and documentation to illustrate the use of a multiline text Param

onestn committed:
docs: Remove outdated 'executor' reference from run() method docstring (#42121)

pierrejeambrun committed:
Fix task_instance and dag_run links from list views (#42138)

bjankie1 committed:
AIP-59-performance-dags (#41961)

bbovenzi committed:
Add Ryan Hamilton as ui codeowner (#42150)

potiuk committed:
Actually move saml to amazon provider (mistakenly added in papermill) (#42148)

Follow up after #42137 -> saml was added mistakenly to papermill,
not amazon :(

jscheffl committed:
Pin airbyte-api to 0.51.0 (#42154) (#42155)

VladaZakharova committed:
Fix vertex AI system tests (#42153)

dependabot[bot] committed:
Bump apache-airflow from 2.10.0 to 2.10.1 in /performance (#42149)

Bumps [apache-airflow](https://github.com/apache/airflow) from 2.10.0 to 2.10.1.
- [Release notes](https://github.com/apache/airflow/releases)
- [Changelog](https://github.com/apache/airflow/blob/main/RELEASE_NOTES.rst)
- [Commits](https://github.com/apache/airflow/compare/2.10.0...2.10.1)

---
updated-dependencies:
- dependency-name: apache-airflow
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

moiseenkov committed:
Exclude partition from BigQuery table name (#42130)

Please enter the commit message for your changes. Lines starting

yuan-glu committed:
Added EA to NTHEWILD.md (#42161)

RNHTTR committed:
document that running task instances will be marked as skipped when a dagrun times out (#41410)

eladkal committed:
Add Airflow 3 development readme (#41457)

* Add Airflow 3 development readme

* fix typo

* Update dev/README_AIRFLOW3_DEV.md

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

* restructure doc

* fix static checks

* updates

* fixes

* fix spell checks

* clarify cherry picking

* add milestones

* fix doc

* Update dev/README_AIRFLOW3_DEV.md

Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

* add 2.11 mental model

* Update dev/README_AIRFLOW3_DEV.md

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>

* fixes

---------

Co-authored-by: Jarek Potiuk <jarek@potiuk.com>
Co-authored-by: Ephraim Anierobi <splendidzigy24@gmail.com>

mayankymailusfedu committed:
Added Aisera to the list of companies using Apache Airflow (#42162)

Added Aisera to INTHEWILD.md

Co-authored-by: Ryan Hatter <25823361+RNHTTR@users.noreply.github.com>

kaydee-edb committed:
Added EnterpriseDB to the list of companies using Airflow (#42163)

althati committed:
Adding my company (#42165)

hikaruhk committed:
Added BTIG to the list of companies using Apache Airflow (#42169)

Co-authored-by: HikaruT <hikaruhk@live.com>
Co-authored-by: Ryan Hatter <25823361+RNHTTR@users.noreply.github.com>

potiuk committed:
Fix wrong casing in airbyte hook. (#42170)

Fixes: #42154

mevich committed:
Added Autodesk to list of companies using Airflow In the wild (#42176)

Co-authored-by: Vikram Medabalimi <vikram.medabalimi@autodesk.com>

ipsatrivedi committed:
Adding Tekmetric to Users list (#42167)

Co-authored-by: Ipsa Trivedi <>

svellaiyan committed:
Added my current company. We use Airflow for production workloads. (#42168)

skandala23 committed:
adding airflowuser for bloomberg (#42173)

* adding airflowuser for bloomberg

* Update INTHEWILD.md

---------

Co-authored-by: Ryan Hatter <25823361+RNHTTR@users.noreply.github.com>

srehman420 committed:
Adding support for volume configurations in ECSRunTaskOperator (#42087)

* Adding support for volume configurations in ECSRunTaskOperator

* Updating template fields override test to include volume_configurations

---------

Co-authored-by: glory9211 <shoaib.rehman9211@gmail.com>

Srabasti committed:
Add Five9 to Airflow Company List (#42172)

utkarsharma2 committed:
Airflow 2.10.1 has been released (#42068)

dirrao committed:
Align timers and timing metrics (ms) across all metrics loggers (#39908)

ipsatrivedi committed:
Updating Tekmetric URL in Airflow User's List (#42179)

* Adding Tekmetric to Users list

* Removing extra space

---------

Co-authored-by: Ipsa Trivedi <>

vfeldsher committed:
Added new user under Bloomberg to INTHEWILD.md (#42178)

leoguzman committed:
reformat summary commands (#42171)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Ryan Hatter <25823361+RNHTTR@users.noreply.github.com>

jscheffl committed:
Adding bare/empty provider package for AIP-69 as starting point (#42046)

bbovenzi committed:
do not camelcase xcom entries (#42182)

dirrao committed:
Logging deprecated configuration removed (#42100)

* Deprecated logging configuration removed

* news fragment added

* Deprecated logging configuration removed

* news fragment added

gopidesupavan committed:
Add template field tests to AWS operators part1 (#42183)

* adding template_fields tests in operators

jscheffl committed:
Add Jens Scheffler for UI as contributor (#42190)

utkarsharma2 committed:
Revert "Handle Example dags case when checking for missing files (#41856)" (#42193)

This reverts commit 435e9687b0c56499bc29c21d3cada8ae9e0a8c53.

sowow359 committed:
Fix require_confirmation_dag_change (#42063)

* Add 'lower' to require_confirmation_dag_change

* Set input attribute only if require-confirmation is true

vincbeck committed:
Move `is_active` user property to FAB auth manager (#42042)

utkarsharma2 committed:
Revert "Fix: DAGs are not marked as stale if the dags folder change (#41433)" (#42197)

This reverts commit 9f30a41874454696ae2b215b2d86cb9a62968006.

jscheffl committed:
Fix problem, skip tests if only not compatible providers are tested a… (#42204)

* Fix problem, skip tests if only not compatible providers are tested against old airflow versions

* Fix pytests

* Fix pytests

bbovenzi committed:
Add extra and renderedTemplates as keys to skip camelCasing (#42206)

* Add extra, conf and renderedTemplates as keys to skip camelCasing

* remove conf, check for extra in lists

detvdl committed:
fix: only treat null/undefined as falsy when rendering XComEntry (#42199)

fix: only treat null, undefined or NaN as NULL in XComEntry render

fix: use coalescing nullish check + isNaN

fix: only treat null/undefined as falsy when rendering XComEntry

pierrejeambrun committed:
Migrate public get dags to fastapi (#42196)

* Migrate public get dags to fastapi

* Fix CI

* Fix CI lowest dependency resolution

atrbgithub committed:
Display rerun icon for tasks (#42152)

This will be displayed for tasks which have been ran more than once.

bbovenzi committed:
Page routing in new UI (#42185)

* Set up ui routing

* Cleanup Nav

* Rename app.tsx to App.tsx

* Rename app.test.tsx to App.test.tsx

* Rename main.tsx to Main.tsx

* Minor changes

* Fix blueprint routes

romsharon98 committed:
fix broken main (#42221)

pavansharma36 committed:
Added infoworks.io in INTOTHEWILD.md (#42212)

potiuk committed:
Fix MyPy Checks introduced by newer version of InfluxDB (#42227)

lucasmo committed:
Fix documentation for cpu and memory usage (#42147)

These metrics are reported in airflow/task/task_runner/standard_task_runner.py without "percent"

romsharon98 committed:
Add pgbouncer and statsd ingress (#41759)

* add pgbouncer and statsd ingress

* convert pgbouncer to PgBouncer

* improve tests, rename flower

* change ports

* fix tests

Rushabh-Garambha committed:
Added Novelis to INTHEWILD.md (#42239)

Co-authored-by: rgarambha <rushabg1@umbc.edu>

MariaAbadi committed:
Add DevITJobs.com to INTHEWILD.md (#42236)

gopidesupavan committed:
Add template fields tests to aws operators (2) (#42202)

o-nikolas committed:
ECSExecutor: Drop params that aren't compatible with EC2 (#42228)

If using the ECS executor with the EC2 launch type, then
platform_version and assign_public_ip are not applicable
and cause the Boto call to fail.

Lee-W committed:
allow dataset alias to add more than one dataset events (#42189)

pierrejeambrun committed:
Fix main otel logger test (#42246)

seyoon-lim committed:
Merge branch 'main' into feature/add-krb-field-on-spark-conn

pierrejeambrun committed:
Add FastAPI migration done marker (#42198)

seyoon-lim committed:
rollback _get_keytab & _get_principal

seyoon-lim committed:
Merge branch 'main' into feature/add-krb-field-on-spark-conn

seyoon-lim committed:
set default values of keytab & principal on conn_data

dirrao committed:
Deprecated configuration removed (#42129)

pierrejeambrun committed:
Update api endpoint documentation for contributors (#42216)

loucyx committed:
UI ESLint (#42184)

* 🚨 new ESLint rules and configuration.

* 🚨 fix all linting errors.

* 🚚 fix component casing.

* 🔧 add new import sorting for prettier.

* 🚨 apply new sorting of imports.

* 🔧 add missing `noUncheckedIndexedAccess` setting to avoid potential mistakes with index access.

* 📄 add license to `prettier.config.js`

* 🎨 format code examples to use spaces instead of tabs.

* 🎨 format openapi-gen files.

* 🔧 exclude `pnpm-lock.yaml` from adding a license.

* 🔧 add ignore of `pnpm-lock.yaml` in more places.
🌐 add statics word.

* 🔧 add `.vscode/settings.json`

* 🔧 add `*.tsbuildinfo` to `.gitignore`.
⬇️ downgrade typescript.

* 🔧 revert `.vscode` files ignore in gitignore.
🔥 remove `settings.json` for now.

* ⏪ revert formatting in .pre-commit-config.yaml

* 🚚 change casing of files to match main.

* 🔀 fix main merge issues -_-

* 🚚 fix incorrect main file casing change.

* 🔥 delete duplicated files that were moved in main.

* 🎨 add missing space in pre-commit-config

* ✏️ fix pnpm-lock.yaml

* 🔧 fix dist ignoring.

* 🤔 Lint and fox vite.config.ts.

jscheffl committed:
Add DB models for Edge Provider (#42047)

seyoon-lim committed:
Merge branch 'main' into feature/add-krb-field-on-spark-conn
